plugins/woocommerce/tests/php/src/Blocks/Shipping/ShippingControllerTest.php

```diff
@@ -90,4 +128,52 @@ function ( $locale ) {
 		// Remove filter.
 		remove_all_filters( 'woocommerce_get_country_locale' );
 	}
+
+	/**
+	 * Test that register_local_pickup handles missing WC()->shipping and other dependencies gracefully.
+	 */
+	public function test_register_local_pickup_also_handles_missing_dependencies() {
+		$wc_backup   = WC();
+		$logger_mock = $this->getMockBuilder( \WC_Logger_Interface::class )->getMock();
+		// We need this to make sure our mock logger is getting returned.
+		add_filter(
+			'woocommerce_logging_class',
+			function () use ( $logger_mock ) {
+				return $logger_mock;
+			},
+			10,
+			0
+		);
+
+		try {
+			// Test that the method does not throw exceptions without missing dependencies.
+			$this->shipping_controller->register_local_pickup();
+			$this->assertTrue( true, 'Method did not throw exceptions without missing dependencies' );
+
+			// Test that the method does not throw exceptions with missing shipping.
+			WC()->shipping = null;
+			$this->shipping_controller->register_local_pickup();
+			$this->assertTrue( true, 'Method did not throw exceptions with missing shipping' );
+
+			// Test that the error is logged when WC()->shipping->register_shipping_method is not available.
+			$logger_mock->expects( $this->once() )
+						->method( 'error' )
+						->with(
+							'Error registering pickup location: WC()->shipping->register_shipping_method is not available',
+							array( 'source' => 'shipping-controller' )
+						);
+
+			// Test that the method does not throw exceptions with missing WC object.
+			global $woocommerce;
+			$incomplete_wc = new \stdClass(); // Object without shipping property.
+			$woocommerce   = $incomplete_wc;
+
+			$this->shipping_controller->register_local_pickup();
+			$this->assertTrue( true, 'Method did not throw exceptions with missing WC object' );
+		} finally {
+			global $woocommerce;
+			$woocommerce = $wc_backup;
+			remove_all_filters( 'woocommerce_logging_class' );
+		}
```

Comment:
Why put this in a try/finally? You can just do it at the end of the test, also the filter removal could go into `tearDown` couldn't it? I think that is a more common pattern, e.g. https://github.com/woocommerce/woocommerce/blob/c9ee77c785f5e319e8b57bc11a765f892b7eb6bf/plugins/woocommerce/tests/php/src/Blocks/StoreApi/Routes/CartItems.php#L93